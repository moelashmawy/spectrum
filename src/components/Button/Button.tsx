import styled from "styled-components";

type Variant = "primary" | "secondary" | "success";

const getButtonColor = (variant?: Variant) => {
  switch (variant) {
    case "primary":
      return "#007bff";
    case "secondary":
      return "#6c757d";
    case "success":
      return "#28a745";
    default:
      return "#007bff";
  }
};

const Button = styled.button<{ variant?: Variant }>`
  padding: 10px;
  cursor: pointer;
  border: none;
  border-radius: 4px;
  color: #fff;
  background-color: ${(props) => getButtonColor(props.variant)};
  transition: background-color 0.3s;

  &:hover {
    opacity: 0.9;
  }
`;

export default Button;
