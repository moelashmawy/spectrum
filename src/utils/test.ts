import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { ReactElement } from "react";

export const createSnapshot = (ComponentToTest: ReactElement) => {
  const {
    container: { firstChild: actual },
  } = render(ComponentToTest);
  expect(actual).toMatchSnapshot();
};
