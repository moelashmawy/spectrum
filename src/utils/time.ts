export const getCurrentTime = () => {
  const d = new Date();
  return d.toLocaleTimeString();
};
