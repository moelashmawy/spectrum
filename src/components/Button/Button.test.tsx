import Button from "./Button";
import { createSnapshot } from "../../utils/test";

describe("Button", () => {
  test("Button snapshot for primary variant", () => {
    createSnapshot(<Button variant="primary">Click me</Button>);
  });

  test("Button snapshot for secondary variant", () => {
    createSnapshot(<Button variant="secondary">Click me</Button>);
  });

  test("Button snapshot for success variant", () => {
    createSnapshot(<Button variant="success">Click me</Button>);
  });
});
