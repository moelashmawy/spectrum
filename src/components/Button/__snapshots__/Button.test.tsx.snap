// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button snapshot for primary variant 1`] = `
<button
  class="sc-bdfCDU fLjGwI"
  variant="primary"
>
  Click me
</button>
`;

exports[`Button Button snapshot for secondary variant 1`] = `
<button
  class="sc-bdfCDU bvppbh"
  variant="secondary"
>
  Click me
</button>
`;

exports[`Button Button snapshot for success variant 1`] = `
<button
  class="sc-bdfCDU iTZJmY"
  variant="success"
>
  Click me
</button>
`;
